src/sidebar/components/test/NotebookView-test.js

```diff
@@ -91,6 +94,20 @@ describe('NotebookView', () => {
     assert.notCalled(fakeLoadAnnotationsService.load);
   });
 
+  it('shows a message if too many annotations to load', () => {
+    // Simulate the loading service emitting an error indicating
+    // too many annotations to load
+    fakeLoadAnnotationsService.load.callsFake(options => {
+      options.onError(new ResultSizeError(5000));
```

Comment:
`onError` needs to be called with a `ResultSizeError` to trigger `NotebookView`'s messaging. Ergo the dependency on `ResultSizeError`. But...hmmm.